### PR TITLE
Initial ApiTestCase

### PR DESCRIPTION
**Pull request by michalmarcinkowski**
Copied class from my workshop for initial review. In order to see usage examples visit  https://github.com/michalmarcinkowski/php-summercamp-2015-rapid-api/tree/master/src/AppBundle/Tests

### CODE REVIEW

File: ApiTestCase.php [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+use Symfony\Component\Finder\Finder;
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Nelmio\Alice\Loader\Yaml as Loader;
+use Symfony\Component\HttpFoundation\Response;
+use Doctrine\ORM\EntityManagerInterface;
+
+abstract class ApiTestCase extends BaseWebTestCase
+{
+    /**
+     * @var \Symfony\Bundle\FrameworkBundle\Client $client
```
Comment (umpirsky) [inline ApiTestCase.php]: No need for leading `\`.

**Comment by umpirsky** [comment]
This is cool, but I still feel like we are missing some useful features like:

```php
$this->assertResponseMatchesJsonType([
    'id' => 'integer',
    'name' => 'string',
    'email' => 'string:email',
    'homepage' => 'string:url|null',
    'created_at' => 'string:date',
    'is_active' => 'boolean'
]);
```

Codeception have this kind of useful methods. :P

Other cool thing is `grabDataFromResponseByJsonPath('$..users[0].id');`

See more on http://codeception.com/docs/modules/REST#.Vk2EJx8SrSc.

File: ApiTestCase.php [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    }
+
+    /**
+     * Gets service from DIC.
+     *
+     * @param $id
+     *
+     * @return object
+     */
+    protected function get($id)
```
Comment (PWalkow) [inline ApiTestCase.php]: maybe getService would be more readable?

File: ApiTestCase.php [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    protected $defaultEntityManager;
+
+    public function __construct()
+    {
+        $this->defaultEntityManager = $this->get('doctrine.orm.default_entity_manager');
+    }
+
+    public function setUp()
+    {
+        $purger = new ORMPurger($this->defaultEntityManager);
```
Comment (PWalkow) [inline ApiTestCase.php]: this one can be very slow... have you considered some memory database and/or fixtures load?

File: ApiTestCase.php [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        $purger->purge();
```
Comment (PWalkow) [inline ApiTestCase.php]: actually database immutability between tests can be achieved by starting transaction on it and rolling-back afterwards.

File: ApiTestCase.php [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    protected $client;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $defaultEntityManager;
+
+    public function __construct()
+    {
+        $this->defaultEntityManager = $this->get('doctrine.orm.default_entity_manager');
```
Comment (PWalkow) [inline ApiTestCase.php]: do we need to know that this is the default one? if some would need another one he/she could get it from container anyway

**Comment by lchrusciel** [comment]
Closed in favour #2

**Pull request closed without merging**